R/parameter_tbl.R

```diff
@@ -0,0 +1,118 @@
+#' Table of epidemiological distributions
+#'
+#' @description This function subsets the epidemiological parameter library to
+#' return only the chosen epidemiological distribution. The results are
+#' returned as a data frame containing the disease, epidemiological
+#' distribution, probability distribution, author of the study, and the year
+#' of publication.
+#'
+#' @inheritParams epidist
+#' @param multi_epidist Either an `<epidist>` object or a list of `<epidist>`
```

Comment:
This argument is called "multiepidist" but can take one epidist. I think it could be confusing.

Reply:
It can take one or multiple `<epidist>` objects. Here's an example where I supply all the Ebola parameters in the database.

``` r
library(epiparameter)
  ed = epidist_db(disease = "Ebola")
#> Returning 17 results that match the criteria (17 are parameterised). 
#> Use subset to filter by entry variables or single_epidist to return a single entry. 
#> To retrieve the citation for each use the 'get_citation' function
```

``` r
  ed
#> # List of 17 <epidist> objects
#> Number of diseases: 1
#> ❯ Ebola Virus Disease
#> Number of epi distributions: 9
#> ❯ hospitalisation to death ❯ hospitalisation to discharge ❯ incubation period ❯ notification to death ❯ notification to discharge ❯ offspring distribution ❯ onset to death ❯ onset to discharge ❯ serial interval
#> [[1]]
#> Disease: Ebola Virus Disease
#> Pathogen: Ebola Virus
#> Epi Distribution: offspring distribution
#> Study: Lloyd-Smith J, Schreiber S, Kopp P, Getz W (2005). "Superspreading and
#> the effect of individual variation on disease emergence." _Nature_.
#> doi:10.1038/nature04153 <https://doi.org/10.1038/nature04153>.
#> Distribution: nbinom
#> Parameters:
#>   mean: 1.500
#>   dispersion: 5.100
#> 
#> [[2]]
#> Disease: Ebola Virus Disease
#> Pathogen: Ebola Virus-Zaire Subtype
#> Epi Distribution: incubation period
#> Study: Eichner M, Dowell S, Firese N (2011). "Incubation period of ebola
#> hemorrhagic virus subtype zaire." _Osong Public Health and Research
#> Perspectives_. doi:10.1016/j.phrp.2011.04.001
#> <https://doi.org/10.1016/j.phrp.2011.04.001>.
#> Distribution: lnorm
#> Parameters:
#>   meanlog: 2.487
#>   sdlog: 0.330
#> 
#> [[3]]
#> Disease: Ebola Virus Disease
#> Pathogen: Ebola Virus-Zaire Subtype
#> Epi Distribution: onset to death
#> Study: The Ebola Outbreak Epidemiology Team, Barry A, Ahuka-Mundeke S, Ali
#> Ahmed Y, Allarangar Y, Anoko J, Archer B, Abedi A, Bagaria J, Belizaire
#> M, Bhatia S, Bokenge T, Bruni E, Cori A, Dabire E, Diallo A, Diallo B,
#> Donnelly C, Dorigatti I, Dorji T, Waeber A, Fall I, Ferguson N,
#> FitzJohn R, Tengomo G, Formenty P, Forna A, Fortin A, Garske T,
#> Gaythorpe K, Gurry C, Hamblion E, Djingarey M, Haskew C, Hugonnet S,
#> Imai N, Impouma B, Kabongo G, Kalenga O, Kibangou E, Lee T, Lukoya C,
#> Ly O, Makiala-Mandanda S, Mamba A, Mbala-Kingebeni P, Mboussou F,
#> Mlanda T, Makuma V, Morgan O, Mulumba A, Kakoni P, Mukadi-Bamuleka D,
#> Muyembe J, Bathé N, Ndumbi Ngamala P, Ngom R, Ngoy G, Nouvellet P, Nsio
#> J, Ousman K, Peron E, Polonsky J, Ryan M, Touré A, Towner R, Tshapenda
#> G, Van De Weerdt R, Van Kerkhove M, Wendland A, Yao N, Yoti Z, Yuma E,
#> Kalambayi Kabamba G, Mwati J, Mbuy G, Lubula L, Mutombo A, Mavila O,
#> Lay Y, Kitenge E (2018). "Outbreak of Ebola virus disease in the
#> Democratic Republic of the Congo, April–May, 2018: an epidemiological
#> study." _The Lancet_. doi:10.1016/S0140-6736(18)31387-4
#> <https://doi.org/10.1016/S0140-6736%2818%2931387-4>.
#> Distribution: gamma
#> Parameters:
#>   shape: 2.400
#>   scale: 3.333
#> 
#> # ℹ 14 more elements
#> # ℹ Use `print(n = ...)` to see more elements.
#> # ℹ Use `parameter_tbl()` to see a summary table of the parameters.
#> # ℹ Explore database online at: https://epiverse-trace.github.io/epiparameter/dev/articles/database.html
```

``` r
  parameter_tbl(ed)
#> # Parameter table:
#> # A data frame:    17 × 7
#>    disease  pathogen epi_distribution prob_distribution author  year sample_size
#>    <chr>    <chr>    <chr>            <chr>             <chr>  <dbl>       <dbl>
#>  1 Ebola V… Ebola V… offspring distr… nbinom            Lloyd…  2005          13
#>  2 Ebola V… Ebola V… incubation peri… lnorm             Eichn…  2011         196
#>  3 Ebola V… Ebola V… onset to death   gamma             The E…  2018          14
#>  4 Ebola V… Ebola V… incubation peri… gamma             WHO E…  2015        1798
#>  5 Ebola V… Ebola V… incubation peri… gamma             WHO E…  2015          49
#>  6 Ebola V… Ebola V… incubation peri… gamma             WHO E…  2015         957
#>  7 Ebola V… Ebola V… incubation peri… gamma             WHO E…  2015         792
#>  8 Ebola V… Ebola V… serial interval  gamma             WHO E…  2015         305
#>  9 Ebola V… Ebola V… serial interval  gamma             WHO E…  2015          37
#> 10 Ebola V… Ebola V… serial interval  gamma             WHO E…  2015         147
#> 11 Ebola V… Ebola V… serial interval  gamma             WHO E…  2015         112
#> 12 Ebola V… Ebola V… hospitalisation… gamma             WHO E…  2015        1167
#> 13 Ebola V… Ebola V… hospitalisation… gamma             WHO E…  2015        1004
#> 14 Ebola V… Ebola V… notification to… gamma             WHO E…  2015        2536
#> 15 Ebola V… Ebola V… notification to… gamma             WHO E…  2015        1324
#> 16 Ebola V… Ebola V… onset to death   gamma             WHO E…  2015        2741
#> 17 Ebola V… Ebola V… onset to discha… gamma             WHO E…  2015        1335
```

<sup>Created on 2024-07-09 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>